sklearn/base.py

```diff
@@ -236,6 +203,61 @@ def set_params(self, **params):
 
         return self
 
+
+class BaseEstimator(GetSetParamsMixin):
+    """Base class for all estimators in scikit-learn
+
+    Notes
+    -----
+    All estimators should specify all the parameters that can be set
+    at the class level in their ``__init__`` as explicit keyword
+    arguments (no ``*args`` or ``**kwargs``).
+    """
+
+    def get_params(self, deep=True):
+        """Get parameters for this estimator.
+
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        try:
+            return super().get_params(deep=deep)
+        except SignatureError as e:
```

Comment:
I think I would be happier with `super().__get_params(deep=deep, name='estimator')` than the introduction of these new exception classes as sentinels.

Reply:
Ok, I'll remove these exception classes and pass the name as an argument.